runtime/lua/vim/ui.lua

```diff
@@ -179,7 +179,13 @@ function M._get_url()
       current_node = current_node:parent()
     end
   end
-  return vim.fn.expand('<cfile>')
+
+  local old_isfname = vim.o.isfname
+  vim.cmd [[set isfname+=@-@]]
+  local url = vim.fn.expand('<cfile>')
```

Comment:
This approach is not quite perfect because L185 can often throw an error:
```
Vim:E446: No file name under cursor
```
in which case `isfname` will never be recovered. We'll probably need to protect call for this kind of approach unless it's guaranteed to not throw.

Reply:
This will now be protected in https://github.com/neovim/neovim/pull/29244 so this won't be a problem after that.